### PR TITLE
Ensure we don't remove endpoints that are in use.

### PR DESCRIPTION
**Pull request by solipet**

### CODE REVIEW

File: _articles/annual-certificate-rotation.md [hunk 1]
```diff
@@ -62,4 +66,28 @@ Partnerships team | [login.gov](http://login.gov/)
 Technology Transformation Services
 U.S. General Services Administration
 [partners@login.gov](mailto:partners@login.gov)
-</blockquote>
\ No newline at end of file
+</blockquote>
+
+
+## Steps to remove and old SAML certificate:
```
Comment (mitchellhenke) [inline _articles/annual-certificate-rotation.md:72]:
```suggestion
## Steps to remove an old SAML certificate:
```

File: _articles/annual-certificate-rotation.md [hunk 2]
```diff
@@ -62,4 +66,28 @@ Partnerships team | [login.gov](http://login.gov/)
...
+1. Run a CloudWatch query to ensure that partners are no longer the endpoint for the year being retired, e.g.,
```
Comment (mitchellhenke) [inline _articles/annual-certificate-rotation.md:73]:
```suggestion
1. Run a CloudWatch query to ensure that partners are no longer using the endpoint for the year being retired:
```

File: _articles/annual-certificate-rotation.md [hunk 3]
```diff
@@ -62,4 +66,28 @@ Partnerships team | [login.gov](http://login.gov/)
...
+    ```
+    fields
+      substr(replace(path, "/api/saml/auth", ""), 0, 4) AS year,
+      @timestamp, @message
+    | sort @timestamp desc
+    | filter path like "/api/saml/auth"
+    | stats count(*) by year
```
Comment (zachmargolis) [inline _articles/annual-certificate-rotation.md:80]: let's comment the file we read from. This is `production.log` right?

Also I was able to get a version of the query that included HTTP referer, let's add this too?

```
# file prod_/var/log/nginx/access.log
fields @timestamp, @message
| parse @message 'http_referer="*"' as http_referer
| parse @message 'uri_path="*"' as path
| sort @timestamp desc
| filter path like "/api/saml/auth"
| filter http_referer not like "secure.login.gov"
| limit 20
| display
  replace(path, "/api/saml/auth", "") AS year,
  http_referer AS http_referer,
  @timestamp
| stats count(*) by year, http_referer
```

File: _articles/annual-certificate-rotation.md [hunk 4]
```diff
@@ -62,4 +66,28 @@ Partnerships team | [login.gov](http://login.gov/)
...
+    ```
+2. If the endpoint is no longer in use, remove reference to the old 20XX cert in secrets saml_endpoint_configs:
```
Comment (mitchellhenke) [inline _articles/annual-certificate-rotation.md:82]:
```suggestion
2. If the endpoint is no longer in use, remove references to the old 20XX cert in secrets saml_endpoint_configs:
```

**Review by mitchellhenke — APPROVED** [review]

**Review by zachmargolis — APPROVED** [review]
Thanks for adding this

File: _articles/annual-certificate-rotation.md (hunk 3 above)
Comment (solipet) [reply _articles/annual-certificate-rotation.md]: Added in f84969e

File: _articles/annual-certificate-rotation.md [hunk 5]
```diff
@@ -5,51 +5,62 @@ layout: article
...
+	 saml_endpoint_configs: '[{"suffix":"2021","secret_key_passphrase":"XXXXXXXXXXXX"},{"suffix":"2022","secret_key_passphrase":"XXXXXXXXXXXX"}]'
 	 ```
-4. Update <https://developers.login.gov/saml/> article to use the new year endpoints
-	- Update <https://developers.login.gov/saml/> article found in [18F/identity-dev-docs](https://github.com/18F/identity-dev-docs) to use the new year certificate.
-	- The endpoints change like this:
-	  - `/api/saml/metadata2019` → `/api/saml/metadata2020`
-	  - `/api/saml/auth2019` → `/api/saml/auth2020`
+    - Recycle the env to make new configs take effect.
+3. Confirm the new endpoint is live by going to `/api/saml/metadata20XX` in that environment's idp.
+3. New SAML 20XX has been tested using a new test app on the Dashboard
```
Comment (orenyk) [inline _articles/annual-certificate-rotation.md:41]: *Nitpick:* I think Markdown can handle it but this should be `4` and the rest of the numbers updated.

**Review by orenyk — APPROVED** [review]
Minor nitpick but otherwise this looks great, thanks for updating it!

**Pull request merged**